bindings/rust/s2n-tls/src/raw/connection.rs

```diff
@@ -80,18 +98,48 @@ impl Connection {
         Ok(self)
     }
 
+    /// Attempts to drop the config on the connection.
+    ///
+    /// # Safety
+    ///
+    /// Manual memory management is error prone and users must not use after free.
```

Comment:
Safety comments are for making general statements. They're for documenting what a caller must ensure before calling this

```suggestion
    /// The caller must ensure the config associated with the connection was created with a [`config::Builder`].
```